### PR TITLE
fix: Submit tx upgrade

### PR DESCRIPTION
**Pull request by PirosB3**
As discussed, we added [takerTokenFillAmount](https://github.com/0xProject/quote-server/pull/21) in the Submit request and response. I've updated the QuoteServerClient to reflect these changes.

### CODE REVIEW

File: src/utils/quote_server_client.ts [hunk 1]
```diff
@@ -28,7 +28,11 @@ export class QuoteServerClient {
             const response = await this._axiosInstance.post(`${makerUri}/submit`, payload, {
                 timeout: ONE_SECOND_MS * 2,
             });
-            schemaValidator.validate(response.data, schemas.submitReceiptSchema);
+            const validator = schemaValidator.validate(response.data, schemas.submitReceiptSchema);
+            if (validator.errors.length > 0) {
```
Comment (PirosB3) [inline src/utils/quote_server_client.ts:32]: @phil-ociraptor `validate` does not throw an error by default, so I just force it to throw an error if errors are present.

File: src/utils/quote_server_client.ts [hunk 2]
```diff
@@ -39,6 +43,18 @@ export class QuoteServerClient {
                 throw new Error('Fee in response is not equal to fee in request');
             }
 
+            if (response.data.signedOrderHash !== payload.orderHash) {
```
Comment (PirosB3) [inline src/utils/quote_server_client.ts:46]: @phil-ociraptor I also added this validation, which makes sure that the right order hash is returned.

**Review by alexkroeger — APPROVED** [review]

File: src/utils/quote_server_client.ts (hunk 1 above)
Comment (phil-ociraptor) [reply src/utils/quote_server_client.ts:32]: hmmm.... I wonder how some of my tests were passing, lol, but all good!

**Review by phil-ociraptor — APPROVED** [review]
Looks good! Tests seem to be failing for a seemingly unrelated test - so I kicked off another build. Once they pass, should be good to go

**Pull request merged**